### ISSUE TITLE
InvalidDomain

### ISSUE THREAD

**Issue by AngelValverdePerez** [underworldcode/underworld2#272]
Hi again,

while running the job in parallel, sometimes it breaks giving me this error message:

File "CantabriaModel_EGU2_SlopeGeomtry.py", line 2953, in <module>
    solver.solve(nonLinearIterate=True, nonLinearTolerance=0.05, nonLinearMaxIterations=40) 
  File "/apps/UNDERWORLD/SRC/python/underworld2-2.2.2b/underworld/systems/_bsscr.py", line 382, in solve
    libUnderworld.StgFEM.SystemLinearEquations_NonLinearExecute(self._stokesSLE._cself, None)
RuntimeError: Invalid domain was encountered while evaluating function.

When this happen, what is going on?

**Comment by jmansour** [comment]
One of your function was being evaluated outside the domain it seems.  Possibly due to a moving mesh?

**Comment by AngelValverdePerez** [comment]
Hi @jmansour 

Not really, my mesh is static now (because remmember I cannot do free surface in parallel). 
I think it was because every step I'm moving the indentor (velocity inicial zone) in an agressive way.

Otherwise, I take this opportunity to tell you about other new error:

File "CantabriaModel_EGU3.py", line 3271, in <module>
    stressMapFn_data = stressMapFn.evaluate(swarm)
  File "/home/avalverde/underworld2-2017.03.14/underworld2/underworld/function/_function.py", line 769, in evaluate
    return _cfn.Query(self._fncself).query(inputData._get_iterator())

What does this error means? First time I saw it

**Comment by jmansour** [comment]
I can't see any error message.

Is it possible you have particles which have left the domain?

For the parallel free surface, did you try the fix I suggested?

**Comment by lmoresi** [comment]
We might consider returning NaN for these situations and encourage people to check their return values.

Sent from my iPhone

On 3 Feb 2018, at 04:28, jmansour <notifications@github.com<mailto:notifications@github.com>> wrote:


I can't see any error message.

Is it possible you have particles which have left the domain?

For the parallel free surface, did you try the fix I suggested?

**Comment by jmansour** [comment]
Yeah there's a few approaches we could take.  I think the current behaviour (raising an exception) is the correct default. 

Where functions are called by `Function.evaluate()`, we could pass in a second argument which is a bool array to be filled which specifies where evaluations have been successful/unsuccessful. Not exactly sure how we'd do this for functions called from other API calls (such as Angel's situation).

**Comment by AngelValverdePerez** [comment]
Hi @jmansour 
I did not check that because my supervisors here told me that we want the sticky air for our models.
And I figured out that I asked for the wrong error, this one is giving me troubles in parallel ....

An uncaught exception was encountered on processor 26.
Traceback (most recent call last):
  File "CantabriaModel_EGUtest.py", line 2502, in <module>
    solver.solve(nonLinearIterate=True, nonLinearTolerance=0.06, nonLinearMaxIterations=20)
  File "/home/avalverde/underworld2-2017.03.14/underworld2/underworld/systems/_bsscr.py", line 382, in solve
    libUnderworld.StgFEM.SystemLinearEquations_NonLinearExecute(self._stokesSLE._cself, None)
RuntimeError: Value overflow error was encountered while evaluating function.


(I have periodic conditions [True,False]) and I'm not deforming the mesh)
